lib/async-git-prompt.zsh

```diff
@@ -0,0 +1,33 @@
+ASYNC_PROC=0
+function precmd() {
+    function async() {
```

Comment:
The function name should probably also be prefixed, like `_omz_git_prompt_async`, since the function namespace is global and flat.